packages/next/src/build/index.ts

```diff
@@ -726,13 +727,28 @@ export default async function build(
         .traceFn(() => loadEnvConfig(dir, false, Log))
       NextBuildContext.loadedEnvFiles = loadedEnvFiles
 
+      const configLoadTrace = new AccessProxy()
+      const configTraceFile = process.env.NEXT_TURBOREPO_TRACE_FILE
+      const isTurborepoTraceEnabled = configTraceFile !== undefined
+      const accessTrace = isTurborepoTraceEnabled
```

Comment:
This pattern is repeated a few times, maybe writing this instead considering you already have a helper?

```ts
import type { AccessProxy } from './access-proxy'

import { withAccessProxy } from './with-access-proxy'

/**
 * Trace access to the filesystem, environment variables, and TCP addresses and
 * merge the results into the parent AccessProxy.
 *
 * @param f the function to trace
 * @param parent the parent AccessProxy to merge the results into
 * @returns the result of the function
 */
export function traceAccess<T>(
  f: () => T | Promise<T>,
  parent: AccessProxy
): Promise<T> | T {
  // If the trace file is not set, don't trace and instead just call the
  // function.
  if (!process.env.NEXT_TURBOREPO_TRACE_FILE) return f()

  // Otherwise, trace the function and merge the results into the parent. Using
  // `then` instead of `await` here to avoid creating a new async context when
  // tracing is disabled.
  return withAccessProxy(f).then(([result, proxy]) => {
    parent.merge(proxy)

    // Return the result of the function.
    return result
  })
}
```

Reply:
Love it!